scripts/background.js

```diff
@@ -145,85 +157,114 @@ function processHeaders(details) {
-            }
-            var storeMatches = tabIds.indexOf(respTabId) >= 0;
-            if (storeMatches) {
-                chrome.cookies.get({"url": url.href, "name": CHL_CLEARANCE_COOKIE, "storeId": store.id}, function(cookie) {
-                    // Require an existing, non-expired cookie.
-                    if (cookie) {
-                        clearanceHeld = (cookie.expirationDate * 1000 >= Date.now());
-                    }
-                });
+    let fired = false;
+    if (CHECK_COOKIES) {
+        chrome.cookies.getAllCookieStores(function(stores) {
+            let clearanceHeld = false;
+            stores.forEach( function(store, index) {
+                var tabIds = store.tabIds;
+                if (tabIds.length && tabIds[0].id !== undefined) {
+                    tabIds = tabIds.map((tab) => respTabId.id);
+                }
+                var storeMatches = tabIds.indexOf(respTabId) > -1;
+                if (storeMatches) {
+                    chrome.cookies.get({"url": url.href, "name": CHL_CLEARANCE_COOKIE, "storeId": store.id}, function(cookie) {
+                        // Require an existing, non-expired cookie.
+                        if (cookie) {
+                            clearanceHeld = (cookie.expirationDate * 1000 >= Date.now());
+                        }
+                    });
+                }
+            });
+
+            // If a clearance cookie is not held then set the spend flag
+            if (!clearanceHeld) {
+                fireRedeem(url, respTabId);
+                fired = true;
             }
         });
+    } else {
+        // If cookies aren't checked then we always attempt to redeem.
+        fireRedeem(url, respTabId);
+        return true;
+    }
+    return fired;
+}
 
-        // If a clearance cookie is not held then set the spend flag
-        if (!clearanceHeld) {
-            setSpendFlag(url.host, true);
-            let targetUrl = target[respTabId];
-            if (url.href == targetUrl) {
-                chrome.tabs.update(respTabId, { url: targetUrl });
-            } else if (!targetUrl || (targetUrl != url.href && !isFaviconUrl(targetUrl))) {
-                // set a reload in the future when the target has been inited
-                futureReload[respTabId] = url.href;
-            }
+// Actually activate the redemption request
+function fireRedeem(url, respTabId) {
+    if (REDEEM_METHOD == "reload") {
+        setSpendFlag(url.host, true);
+        let targetUrl = target[respTabId];
+        if (url.href == targetUrl) {
+            chrome.tabs.update(respTabId, { url: targetUrl });
+        } else if (!targetUrl || (targetUrl != url.href && !isFaviconUrl(targetUrl))) {
+            // set a reload in the future when the target has been inited
+            futureReload[respTabId] = url.href;
         }
-    });
+    } else {
+        throw new Error("[privacy-pass]: Incompatible redeem method selected.");
```

Comment:
For now, I think specifying that reload is the only supported REDEEM_METHOD at the moment could be useful:
```suggestion
        throw new Error("[privacy-pass]: Incompatible redeem method selected. Only \"reload\" is supported at the moment.");
```